src/npc.c

```diff
@@ -181,14 +660,55 @@ void update_npc_blur(Npc* npc) {
 
 INCLUDE_ASM(void, "npc", appendGfx_npc_blur, Npc* npc);
 
-INCLUDE_ASM(s32, "npc", func_8003B184);
+void func_8003B184(void) {
+    D_800A0B94 = 1;
+}
 
-INCLUDE_ASM(s32, "npc", func_8003B198);
+void func_8003B198(void) {
+    D_800A0B94 = 0;
+}
 
 void func_8003B1A8(void) {
 }
 
-INCLUDE_ASM(s32, "npc", func_8003B1B0);
+s32* func_802DEA40(u16);
+
+void func_8003B1B0(void) {
+    s32 i;
+    s32 j;
+
+    for (i = 0; i < MAX_NPCS; i++) {
+        Npc* npc = (*gCurrentNpcListPtr)[i];
+        if (npc != NULL) {
+            if (npc->flags && !(npc->flags & 0x40000000)) {
+                if (!(npc->flags & 0x1000000)) {
+                    if (!(npc->flags & 0x4000000)) {
+                        npc->spriteInstanceID = func_802DE0EC(npc->currentAnim.w, npc->extraAnimList);
+                    } else {
+                        npc->spriteInstanceID = func_802DE0EC(npc->currentAnim.w | 0x80000000, npc->extraAnimList);
+                    }
+                }
+                if (!(npc->flags & 0x40000000)) {
+                    if (!(npc->flags & 0x1000000) && (npc->unk_B4 != 0)) {
+                        npc->spritePaletteList = func_802DEA40(npc->currentAnim.h);
+                        for (npc->paletteCount = 0; npc->spritePaletteList[npc->paletteCount] != -1; npc->paletteCount++) {
+
```

Comment:
I think a comment would be nice here mentioning why there's an empty loop / what's happening here